### DISCUSSION TITLE
Current season summary

### DISCUSSION THREAD

**Discussion by elierpf** [tropycal/tropycal#201] [Q&A]
Hello and thanks in advance:
There is any way to plot a current season summary,  with the preliminary track of all storms?

**Comment by tomerburg** [comment]
Yes - to do so you'd have to retrieve a TrackDataset object including the current season. For the Atlantic & East Pacific basins, you'd need to include a ``include_btk = True`` flag to ensure you're reading in the current season, then plot a summary plot for the season:

```python
from tropycal import tracks

# Read in this year's data for the North Atlantic basin
basin = tracks.TrackDataset(basin='north_atlantic', include_btk=True)

# Make a summary plot
basin.plot_season(2023)
```

For other global basins, you'd need to use IBTrACS, which some methods of reading IBTrACS (e.g., JTWC) include the latest season with a couple day lag time. Here's an example for the West Pacific:

```python
from tropycal import tracks

# Read in this year's data for the West Pacific basin using JTWC data
basin = tracks.TrackDataset(basin='west_pacific', ibtracs_mode='jtwc')

# Make a summary plot
basin.plot_season(2023)
```
